### ISSUE TITLE
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: UnexpectedEof, error: StringError("failed to fill whole buffer") }', libcore/result.rs:1009:5

### ISSUE THREAD

**Issue by bjorn3** [mdsteele/rust-ar#9]
This is from an unwrap on the result of `archive.next_entry()`. `ar` can read it just fine.

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: UnexpectedEof, error: StringError("failed to fill whole buffer") }', libcore/result.rs:1009:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:59
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: rustc::util::common::panic_hook
   5: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:480
   6: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   7: rust_begin_unwind
             at libstd/panicking.rs:325
   8: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   9: core::result::unwrap_failed
             at /rustc/6b9b97bd9b704f85f0184f7a213cc4d62bd9654c/src/libcore/macros.rs:26
  10: <core::result::Result<T, E>>::unwrap
             at /rustc/6b9b97bd9b704f85f0184f7a213cc4d62bd9654c/src/libcore/result.rs:808
```

**Comment by bjorn3** [comment]
[abc.zip](https://github.com/mdsteele/rust-ar/files/2590729/abc.zip)

**Comment by bjorn3** [comment]
If you have [evcxr](https://github.com/google/evcxr) installed you can easily repro using this. (Assuming you put the extracted abc.a file in the current dir)

```rust
$ evcxr
Welcome to evcxr. For help, type :help
>> :dep ar="0.6.1"
>> extern crate ar;
>> let mut archive = ar::Archive::new(std::fs::File::open("abc.a").unwrap());
>> while let Some(entry) = archive.next_entry() {entry.unwrap();}
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: UnexpectedEof, error: StringError("failed to fill whole buffer") }', libcore/result.rs:1009:5
stack backtrace:
[...]
```

If you don't have it installed create a new cargo project add `ar` as dependency and put this in main.rs:

```rust
extern crate ar;

fn main() {
    let mut archive = ar::Archive::new(std::fs::File::open("abc.a").unwrap());
    while let Some(entry) = archive.next_entry() {entry.unwrap();}
}
```

**Comment by bjorn3** [comment]
This likely comes from https://github.com/mdsteele/rust-ar/blob/7cfb6dc04b492296653c8853e320f78882b3474f/src/lib.rs#L236

**Comment by mdsteele** [comment]
Hi, thanks very much for the detailed report, and sorry for the slow reply.  I am able to reproduce the error.  I will try to dig into what's going on here.

**Comment by bjorn3** [comment]
> sorry for the slow reply

no problem

> I will try to dig into what's going on here.

thanks

**Comment by bjorn3** [comment]
Thanks. I unfortunately can't remember the original way I found this bug so I can't test it. :( Could you publish a new version.

**Comment by mdsteele** [comment]
Sure, published as [v0.6.2](https://crates.io/crates/ar/0.6.2).
